e2e/src/simple-token-transfer.spec.ts

```diff
@@ -0,0 +1,172 @@
+const {Assertion, expect} = require("chai");
+import { grpc } from "orbs-common-library/src/grpc";
+import { types } from "orbs-common-library/src/types";
+import { CryptoUtils } from "orbs-common-library/src/CryptoUtils";
+import { delay } from "bluebird";
+
+
+const accounts = new Map<string, Account>();
+
+type contractMethodArgs = [string | number] | undefined[];
+
+class HardCodedContactClient {
+    orbsClient: OrbsClient;
+
+    constructor(orbsClient: OrbsClient) {
+        this.orbsClient = orbsClient;
+    }
+
+    protected async sendTransaction(methodName: string, args: contractMethodArgs) {
+        const argumentsJson = JSON.stringify({
+            method: methodName,
+            args: args
+        });
+        return this.orbsClient.sendTransaction(BarContractClient.BAR_CONTRACT_ADDRESS, argumentsJson);
+    }
+
+    protected async call(methodName: string, args: contractMethodArgs) {
+        const argumentsJson = JSON.stringify({
+            method: methodName,
+            args: args
+        });
+        return this.orbsClient.call(BarContractClient.BAR_CONTRACT_ADDRESS, argumentsJson);
+    }
+}
+
+class BarContractClient extends HardCodedContactClient {
+    static BAR_CONTRACT_ADDRESS = "barbar";
+
+    public initBalance(account: string, balance: number) {
+        return this.sendTransaction("initBalance", [account, balance]);
+    }
+
+    public transfer(to: string, amount: number) {
+        return this.sendTransaction("transfer", [to, amount]);
+    }
+
+    public getMyBalance() {
+        return this.call("getMyBalance", []);
+    }
+}
+
+class OrbsClient {
+    crypto: CryptoUtils;
```

Comment:
It reminds me: I'll make sure that our CryptoUtils is a "native" library, supporting everything we might need (probably ed25519, SHA3, RIPEMD160 for now).

Reply:
In general I'd avoid naming anything `FooUtils` or `CommonFoo`, as it lends legitimacy to these pieces of code becoming rubbish bins. Try to decide what does it do and name it accordingly. 

See here: http://arlobelshee.com/good-naming-is-a-process-not-a-single-step/

Reply:
Agreed, but it's not related to this specific pull request.